### PR TITLE
PERF-3129 Work around the issue of `ISODate` type and fix other minor issues

### PR DESCRIPTION
**Pull request by yun-soo**
This PR is to work around the issue of `ISODate` in the yaml file by introducing a parameterized configuration `GetBsonDate`. Also added column store index bulk builder metric and request `explain` with `executionStats` instead of the plain `aggregate` command.

Companion PR:
- https://github.com/10gen/dsi/pull/1062

Evergreen patch build:
https://spruce.mongodb.com/version/62bd2e520ae606411b5eec73/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

### CODE REVIEW

File: src/workloads/execution/ColumnStoreIndex.yml [hunk 1]
```diff
@@ -302,51 +372,84 @@ TopTenantsByNoOfEventsQuery: &TopTenantsByNoOfEventsQuery
...
     ]
-  cursor: {batchSize: 50000}
+  cursor: {}
 
 # Defines test operations.
 TestOperations: &TestOperations
 - OperationMetricsName: CountPerActionQuery
   OperationName: RunCommand
-  OperationCommand: *CountPerActionQuery
+  OperationCommand:
```
Comment (dalyd) [inline src/workloads/execution/ColumnStoreIndex.yml:381]: This doesn't look like ```Fix use of ISODate in column store genny workload``` Should this be in a separate PR, or should we update the ticket? 

Also, could you explain in a comment why we are using explain here.

Comment (yun-soo) [reply src/workloads/execution/ColumnStoreIndex.yml:384]: According to the description, it meant working around the issue. Updated the title of the ticket. Also added the following comment.

```
# Here we use explain with verbosity = execustionStats instead of using plain aggregate command
# so that aggregate pipeline runs to completion but yet we don't need to exhaust cursor. As of now
# we don't have an easy way to exhaust cursor.
```

**Review by yun-soo — COMMENTED** [review]
Thanks for review, David! Addressed your comments.

File: src/phases/execution/GetBsonDate.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query-execution"
```
[the remark this answers is not in the document]
Comment (dalyd) [reply src/phases/execution/GetBsonDate.yml:2]: Sounds good. Maybe add a TODO with a link to the ticket somewhere.

**Review by dalyd — DISMISSED** [review]

File: src/phases/execution/GetBsonDate.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (yun-soo) [reply src/phases/execution/GetBsonDate.yml:2]: Done.

**Review by yun-soo — COMMENTED** [review]
Thanks again for review, David! Added a TODO

**Review by dalyd — APPROVED** [review]
